barretenberg/cpp/src/barretenberg/commitment_schemes/shplonk/shplemini.hpp

```diff
@@ -325,41 +326,40 @@ template <typename Curve> class ShpleminiVerifier_ {
      * @param scalars Output vector where the computed scalars will be stored.
      * @param constant_term_accumulator The accumulator for the summands of the constant term.
      */
-    static void batch_gemini_claims_received_from_prover(const size_t log_circuit_size,
-                                                         const std::vector<Commitment>& fold_commitments,
-                                                         const std::vector<Fr>& gemini_evaluations,
-                                                         const std::vector<Fr>& inverse_vanishing_evals,
-                                                         const Fr& shplonk_batching_challenge,
-                                                         std::vector<Commitment>& commitments,
-                                                         std::vector<Fr>& scalars,
-                                                         Fr& constant_term_accumulator)
+    static void batch_gemini_claims_received_from_prover(
+        const size_t log_circuit_size,
+        const std::vector<Commitment>& fold_commitments,
+        const std::vector<Fr>& gemini_evaluations,
+        [[maybe_unused]] const std::vector<Fr>& inverse_vanishing_evals,
+        const Fr& shplonk_batching_challenge,
+        std::vector<Commitment>& commitments,
+        std::vector<Fr>& scalars,
+        Fr& constant_term_accumulator)
     {
 
         // Initialize batching challenge as ν²
         Fr current_batching_challenge = shplonk_batching_challenge.sqr();
         for (size_t j = 0; j < CONST_PROOF_SIZE_LOG_N - 1; ++j) {
-            bool is_dummy_round = j >= (log_circuit_size - 1);
             // Compute the scaling factor  (ν²⁺ⁱ) / (z + r²⁽ⁱ⁺²⁾) for i = 0, … , d-2
             Fr scaling_factor = current_batching_challenge * inverse_vanishing_evals[j + 2];
 
+            // Add Aᵢ(−r²ⁱ) for i = 1, … , n-1 to the constant term accumulator
+            constant_term_accumulator += scaling_factor * gemini_evaluations[j + 1];
+
+            // Update the batching challenge
+            current_batching_challenge *= shplonk_batching_challenge;
+
             if constexpr (Curve::is_stdlib_type) {
                 auto builder = shplonk_batching_challenge.get_context();
                 // TODO(https://github.com/AztecProtocol/barretenberg/issues/1114): insecure!
-                stdlib::bool_t dummy_round = stdlib::bool_t(builder, is_dummy_round);
+                stdlib::bool_t dummy_round = stdlib::witness_t(builder, j >= (log_circuit_size - 1));
                 Fr zero = Fr(0);
-                zero.convert_constant_to_fixed_witness(builder);
                 scaling_factor = Fr::conditional_assign(dummy_round, zero, scaling_factor);
```

Comment:
Initially, I thought that making the bool_t a constant is a simplification and alright for now so I don't make a witness out of thin air (note that for this code to be somewhat secure bool_t has to be a witness tied to the `circuit_size` Given the fact that circuit size i received from transcript, it will itself be a witness). The problem with having `dummy_round` be a constant is that the logic of `conditional_assign` will result in either returning lhs or rhs based on whether dummy_round is true which changes the sigma polynomials. On the other hand, with dummy_round being a witness the scaling_factor, after applying `conditional_assign` becomes a new witness which is the output of additional gates.

Reply:
thanks for explaining this!